.github/workflows/ci.yaml

```diff
@@ -46,26 +46,30 @@ jobs:
         tanzu plugin install builder --local ${TANZU_HOME}/admin-plugins
         tanzu plugin install test --local ${TANZU_HOME}/admin-plugins
         tanzu plugin list
-    - name: Test
-      run: make test
     - name: Scan Inclusive Terminology
       uses: get-woke/woke-action@v0
       with: 
         fail-on-error: true
         woke-args: -c https://via.vmw.com/its-woke-rules
+    - name: Test
+      run: make test
     - name: Build Version
       run: echo "BUILD_VERSION=${GITHUB_REF/refs\/tags\//}" >> $GITHUB_ENV
       if: startsWith(github.ref, 'refs/tags/')
     - name: Build
-      run: |
-        make publish
+      run: make build
+    - name: Publish
+      run: make publish
     - uses: actions/upload-artifact@v3
       with:
         name: tanzu-apps-plugin.tar.gz
         path: tanzu-apps-plugin*.tar.gz
         retention-days: 1
     - name: Install
-      run: make install
+      run: |
+        BUILD_SHA_SHORT=$(git rev-parse --short HEAD)
```

Comment:
For normal CI run(without tags) , this flow works because `BUILD_VERSION` in previous step is not set. When tag is cut, `BUILD_VERSION` will be set so how does installing plugin which is built with one version and running command with another version succeed?

Reply:
Updated commands to validate if BUILD_VERSION is preset before reassigning the version.